javascript/code-challenges/challenges-07.test.js

```diff
@@ -0,0 +1,359 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function called sortStarWarsCharacters that sorts the characters in the starWarsPeople array by height from tallest to shortest.
+------------------------------------------------------------------------------------------------ */
+
+let starWarsPeople = [
+  {
+    "name": "C-3PO",
+    "height": "167",
+    "eye_color": "yellow"
+  },
+  {
+    "name": "Luke Skywalker",
+    "height": "172",
+    "eye_color": "blue"
+  },
+  {
+    "name": "R2-D2",
+    "height": "96",
+    "eye_color": "red"
+  }
+];
+
+const sortStarWarsCharacters = (starWarsArr) => {
+  return starWarsPeople.sort((a, b) => b.height - a.height);
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named removeThree that takes an index and an array. The function should removes three items in the array starting with the value at the index. 
+------------------------------------------------------------------------------------------------ */
+
+const removeThree = (idx, arr) => {
+  return arr.splice(0, 3);
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named joinArray that takes an array and joins all of the elements together in one string on a space.
+------------------------------------------------------------------------------------------------ */
+
+const joinArray = (arr) => {
+  return arr.join(' ');
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named howMuchPencil that takes in a string, as written on the side of a pencil.
+
+As you sharpen the pencil, the string will become shorter and shorter, starting by removing the first letter.
+
+Your function should use slice within a loop and return an array of each successive string result from losing letters to the sharpener, until nothing is left.
+
+For example, if the input is 'Welcome', the output will be:
+['Welcome', 'elcome', 'lcome', 'come', 'ome', 'me', 'e', ''].
+------------------------------------------------------------------------------------------------ */
+
+const howMuchPencil = (str) => {
+  let result = [];
+  for (let i = str.length; i <= 0; i++) {
+    result += str[i];
+    result.slice(i);
```

Comment:
I assume this isn't passing because, unfortunately, you can't just add to an array using +. You'll need to either use push, or shift, depending on where you want to add it to (there are other methods too but...)
You also don't want to slice the entire array, you want to slice individual bits of data WITHIN the array. So, you'd want to slice parts of str at a specific index, and then add that (push) to the result array.